langchain4j-core/src/main/java/dev/langchain4j/http/HttpClientBuilder.java

```diff
@@ -0,0 +1,33 @@
+package dev.langchain4j.http;
+
+import dev.langchain4j.Experimental;
+
+import java.time.Duration;
+
+@Experimental
+public interface HttpClientBuilder {  // TODO move inside HttpClient?
```

Comment:
Looks like a properties object: `OkHttpHttpClientProperties`

I would even delegate this to concrete factory implementation: it should take care of timeouts